Bug #18356: Xen integration for UCS

Check for xenfs.
Create loop-back-devices.
Add usplash support via LSB functions.
--- a/tools/hotplug/Linux/init.d/xend	2010-03-26 17:15:36.000000000 +0100
+++ b/tools/hotplug/Linux/init.d/xend	2010-03-26 17:15:42.000000000 +0100
@@ -22,16 +22,26 @@
 if   test "x$1" = xstart && \
      test -d /proc/xen && \
    ! test -d /proc/xen/capabilities && \
-     grep '	xenfs$' /proc/filesystems >/dev/null && \
    ! grep '^xenfs ' /proc/mounts >/dev/null;
 then
 	mount -t xenfs xenfs /proc/xen
 fi
 
+. /lib/lsb/init-functions
+
 if ! grep -q "control_d" /proc/xen/capabilities ; then
 	exit 0
 fi
 
+
+# usage: makedev name [bcu] major minor owner group mode
+makedev ()
+{
+	mknod "$1" "$2" "$3" "$4"
+	chown "$5":"$6" "$1"
+	chmod "$7" "$1"
+}
+
 # Wait for Xend to be up
 function await_daemons_up
 {
@@ -48,14 +58,32 @@
 
 case "$1" in
   start)
+	log_daemon_msg "Starting XEN control daemon"
+	# We need /dev/loop0 - /dev/loop7 for mounting interfaces
+	# Test, whether old-style loopback devices exist by checking for /dev/loop0,
+	# udev creates /dev/loop/0
+
+	if [ ! -e /dev/loop0 ]; then
+		cd /dev
+		disk="  root disk   0660"
+		for part in 0 1 2 3 4 5 6 7; do
+	        	makedev loop$part b 7 "$part" $disk
+		done
+		cd -
+	fi
+
+
 	mkdir -p /var/lock/subsys
 	touch /var/lock/subsys/xend
 	xend start
 	await_daemons_up
+	log_end_msg 0
 	;;
   stop)
+	log_daemon_msg "Stopping XEN control daemon"
 	xend stop
 	rm -f /var/lock/subsys/xend
+	log_end_msg 0
 	;;
   status)
 	xend status
@@ -64,8 +92,10 @@
         xend reload
         ;;
   restart|force-reload)
+	log_daemon_msg "Restarting XEN control daemon"
 	xend restart
 	await_daemons_up
+	log_end_msg 0
 	;;
   *)
 	# do not advertise unreasonable commands that there is no reason
